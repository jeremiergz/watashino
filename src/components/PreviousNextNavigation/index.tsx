import Link from 'components/Link';
import Box from 'components/primitives/Box';
import FlexBox, { FlexBoxProps } from 'components/primitives/FlexBox';
import React from 'react';

const PreviousNextNavigation: React.FC<PreviousNextNavigationProps> = ({ next, previous, ...rest }) => {
  const shouldDisplay = !!previous || !!next;
  return (
    shouldDisplay && (
      <FlexBox justifyContent="space-between" {...rest}>
        {previous ? (
          <Link aria-label="Previous article" fontSize={16} fontWeight="semiBold" to={previous} variant="underlined">
            {'Prev'}
          </Link>
        ) : (
          <Box />
        )}
        {next ? (
          <Link aria-label="Next article" fontSize={16} fontWeight="semiBold" to={next} variant="underlined">
            {'Next'}
          </Link>
        ) : (
          <Box />
        )}
      </FlexBox>
    )
  );
};

PreviousNextNavigation.displayName = 'PreviousNextNavigation';

export type PreviousNextNavigationProps = FlexBoxProps & {
  next: string;
  previous: string;
};
export default PreviousNextNavigation;
