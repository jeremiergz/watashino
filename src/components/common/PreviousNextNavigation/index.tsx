import React from 'react';
import Box from '../../primitives/Box';
import Flex, { FlexProps } from '../../primitives/Flex';
import Link from '../Link';

type PreviousNextNavigationProps = FlexProps & {
  next: string;
  previous: string;
};

const PreviousNextNavigation = ({ next, previous, ...rest }: PreviousNextNavigationProps) => {
  const shouldDisplay = !!previous || !!next;
  return (
    shouldDisplay && (
      <Flex justifyContent="space-between" {...rest}>
        {previous ? (
          <Link fontSize={16} fontWeight="semi-bold" to={previous} variant="underlined">
            {'Prev'}
          </Link>
        ) : (
          <Box />
        )}
        {next ? (
          <Link fontSize={16} fontWeight="semi-bold" to={next} variant="underlined">
            {'Next'}
          </Link>
        ) : (
          <Box />
        )}
      </Flex>
    )
  );
};

PreviousNextNavigation.displayName = 'PreviousNextNavigation';

export { PreviousNextNavigationProps };
export default PreviousNextNavigation;
