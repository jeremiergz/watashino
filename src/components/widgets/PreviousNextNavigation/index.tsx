import React from 'react';
import Box from '../../primitives/Box';
import Flex, { FlexProps } from '../../primitives/Flex';
import Link from '../Link';

type PreviousNextNavigationProps = FlexProps & {
    next: string;
    previous: string;
};

const PreviousNextNavigation = ({ next, previous, ...rest }: PreviousNextNavigationProps) => {
    return (
        <Flex justifyContent="space-between" {...rest}>
            {previous ? <Link to={previous}>{'◂ Prev'}</Link> : <Box />}
            {next ? <Link to={next}>{'Next ▸'}</Link> : <Box />}
        </Flex>
    );
};

PreviousNextNavigation.displayName = 'PreviousNextNavigation';

export { PreviousNextNavigationProps };
export default PreviousNextNavigation;
