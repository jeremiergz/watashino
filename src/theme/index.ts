const breakpoints = ['320px', '375px', '425px', '768px', '1024px', '1200px', '1440px', '2560px'] as const;
Object.defineProperties(breakpoints, {
  mobileS: { value: breakpoints[0] },
  mobileM: { value: breakpoints[1] },
  mobileL: { value: breakpoints[2] },
  tablet: { value: breakpoints[3] },
  laptopS: { value: breakpoints[4] },
  laptopM: { value: breakpoints[5] },
  laptopL: { value: breakpoints[6] },
  desktop: { value: breakpoints[7] },
});

const colors = {
  black: '#212121',
  dark: '#424242',
  error: '#F44336',
  gray: '#757575',
  info: '#2196F3',
  lightgray: '#E0E0E0',
  primary: '#31859A',
  secondary: '#17365C',
  success: '#4CAF50',
  white: '#FAFAFA',
} as const;

const fonts = {
  main: "'Open Sans', sans-serif",
} as const;

const fontWeights = {
  bold: 800,
  regular: 400,
  'semi-bold': 600,
} as const;

const DarkTheme = {
  background: colors.black,
  breakpoints,
  colors: {
    ...colors,
    text: colors.white,
  },
  fonts,
  fontWeights,
  svg: colors.white,
  type: 'dark',
};

const LightTheme = {
  background: colors.white,
  breakpoints,
  colors: {
    ...colors,
    text: '#262626',
  },
  fonts,
  fontWeights,
  svg: '#262626',
  type: 'light',
};

type Theme = typeof DarkTheme | typeof LightTheme;
type ThemeProps = {
  background?: string;
  colors?: keyof Theme['colors'] | 'inherit';
  fontWeight?: keyof Theme['fontWeights'];
  svg?: string;
  type?: 'light' | 'dark';
};

export { colors, DarkTheme, LightTheme, Theme, ThemeProps };
