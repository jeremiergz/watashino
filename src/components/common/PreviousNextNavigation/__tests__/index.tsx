import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { DarkTheme } from '../../../../theme';
import PreviousNextNavigation from '../../PreviousNextNavigation';

describe('components/widgets/PreviousNextNavigation Test Suite', () => {
  it('matches snapshot', () => {
    const jsx = (
      <ThemeProvider theme={DarkTheme}>
        <PreviousNextNavigation next="/next" previous="/previous">
          PreviousNextNavigation Component
        </PreviousNextNavigation>
      </ThemeProvider>
    );
    expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
  });
});
