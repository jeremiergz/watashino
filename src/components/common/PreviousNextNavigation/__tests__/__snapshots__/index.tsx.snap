// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/widgets/PreviousNextNavigation Test Suite matches snapshot 1`] = `
<div
  className="sc-bdVaJa drumYD"
  display="flex"
>
  <a
    className="sc-bdVaJa sc-bwzfXH ljCIEc"
    href="/previous"
    target="_blank"
  >
    Prev
  </a>
  <a
    className="sc-bdVaJa sc-bwzfXH exmAQC"
    href="/next"
    target="_blank"
  >
    Next
  </a>
</div>
`;
